### DISCUSSION TITLE
Simplify call from ribbon

### DISCUSSION THREAD

**Discussion by josef-poetzl** [joyfullservice/msaccess-vcs-addin#489] [Ideas]
Is there a specific reason for the file name "Version Control.accda" (with spaces and no identifying string)?
If the name did not contain a space, it would be much easier to call up via the ribbon. I would use the name _msaccess-vcs.accda_.

You could use `Application.Run "{Path}\{FileName without extension}.HandleRibbonCommand"` directly and would not have to make sure that the appropriate VBProject is active in order to be able to call `HandleRibbonCommand`.

Example code: https://github.com/josef-poetzl/msaccess-vcs-addin/blob/Simplify-call-from-ribbon/Ribbon/Source/Sources/AddInRibbon.twin#L151

**Comment by joyfullservice** [comment]
Hi Josef,

Thanks for sharing this suggestion! You are right that this syntax of calling the add-in name _without the extension_, followed by the public function both loads the add-in and calls the function. This was something [I tried to figure out in the past](https://stackoverflow.com/questions/62270088), but the closest I had gotten was the call to load the add-in. Thank you so much for sharing this tip!!

In my testing, I am able to successfully call the add-in even with a space in the file name. I have tested this on both accdb and adp projects in Access 2010 and it works great!
```vba
GetObject(,"Access.Application").run Environ$("AppData") & "\MSAccessVCS\Version Control.HandleRibbonCommand", "btnShow"

```

I will plan to update the code base to transition to this simpler and cleaner approach to calling the add-in.

Also, I don't think I quite understood your comment about "no identifying string" in the file name... Were you referring to the fact that "Version Control" is a very generic name? I like your suggestion of `msaccess-vcs.accda` although if the space in the name is not actually a technical problem, there is less of a compelling reason to change it. 🤔 I like the name `Version Control.accda` in that it is very clear that we are talking about an add-in that relates to version control. It uses its own dedicated `MSAccessVCS` folder in AppData, and you see both names in the VBA IDE, so it seems to me that you get the advantage of both names in most contexts.

![image](https://github.com/joyfullservice/msaccess-vcs-addin/assets/7254992/2aad4d11-6a9e-4ea6-a0ab-aca6fb5880e1)

P.S.
Your AccUnit project has been my [favorite](https://stackoverflow.com/questions/3569362) unit testing framework for VBA. The only drawback was the COM dependencies. Although reviewing this further today, it looks like this has been improved quite a bit in the last couple years. I will definitely have to take a closer look.  👍

**Reply by josef-poetzl** [reply]
Hi!

> In my testing, I am able to successfully call the add-in even with a space in the file name. 

I thought that might be a problem - at least that's what I remembered. But I've just tried it out and it also works with a space.

> Were you referring to the fact that "Version Control" is a very generic name?

That's right. But perhaps my thoughts are too focused on add-ins that are located in the MS add-in directory. Since you are using a separate directory, the uniqueness is given again.
"Version Control" is probably even more recognizable to many than VCS. This abbreviation may not be familiar to many Access developers. Our goal must be to get more and more people to know and, above all, use it. :)
